tripy/tripy/backend/api/executable.py

```diff
@@ -221,11 +226,16 @@ def add(a, b):
             compiled_add = tp.compile(add, args=[tp.InputInfo(([1, 2, 3],), dtype=tp.float32), tp.InputInfo(([1, 2, 3],), dtype=tp.float32)])
             print(compiled_add.get_output_info())
         """
-        output_info = []
-        offset = self._executable_signature.get_num_input_args()
-        for idx in range(self._executable_signature.get_num_output_args()):
-            output_info.append(self._get_arg_info(idx + offset))
-        return output_info
+        num_input_args = self._executable_signature.get_num_input_args()
+        num_output_args = self._executable_signature.get_num_output_args()
+        num_results = self._executable_signature.get_num_results()
+
+        assert not (num_output_args and num_results), "Cannot have both output arguments and results"
+
+        if num_output_args:
+            return [self._get_arg_info(idx + num_input_args) for idx in range(num_output_args)]
+        else:
+            return [self._get_result_info(idx) for idx in range(num_results)]
```

Comment:
Why don't we just return an empty list when there are no outputs?

Reply:
I guess what is the distinction between output vs. result?

Reply:
For Tripy, we are only returning results, so we could get rid of output arguments. MLIR-TRT supports function which both accept output arguments and return results.
```
func add(int a, int b, int* r) {
    *r = a + b;
    return *r;
}
```

Reply:
`num_output_args` check is preserving old path. I will get rid of it.